manga_translator/translators/vertex.py

```
@@ -0,0 +1,586 @@
+        last_exception = None
+        local_attempt = 0  # 本次批次的尝试次数
+
+        while is_infinite or attempt < max_retries:
+            # 检查是否被取消
+            self._check_cancelled()
+
+            # 检查全局尝试次数
+            if not self._increment_global_attempt():
+                self.logger.error("Reached global attempt limit. Stopping translation.")
+                last_error_msg = str(last_exception) if last_exception else "Unknown error"
+                raise Exception(f"达到最大尝试次数 ({self._max_total_attempts})，最后一次错误: {last_error_msg}")
```

Ruff (0.14.13)

Create your own exception

(TRY002)